src/iam.ts

```diff
@@ -262,14 +266,37 @@ export class IAM extends IAMBase {
   async updateDidDocument(options: {
     didAttribute: DIDAttribute;
     data: IUpdateData;
+    did?: string;
     validity?: number;
   }): Promise<boolean> {
-    const { didAttribute, data, validity } = options;
-    if (this._document) {
+    const { didAttribute, data, validity, did } = options;
+
+    if (!this._document) {
+      throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
+    }
+
+    if (!this._didSigner) {
+      throw new Error(ERROR_MESSAGES.SIGNER_NOT_INITIALIZED);
+    }
+
+    if (!did) {
       const updated = await this._document.update(didAttribute, data, validity);
       return Boolean(updated);
     }
-    throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
+
+    const defaultValue: IAttributePayload = { publicKey: `0x${new Keys().publicKey}` };
+
+    const updateData: IUpdateData = {
+      algo: Algorithms.Secp256k1,
+      encoding: Encoding.HEX,
+      ...data,
+      value: { ...defaultValue, ...data.value }
+    };
+
+    const operator = new ProxyOperator(this._didSigner, this._registrySetting, addressOf(did));
+    const update = await operator.update(did, DIDAttribute.PublicKey, updateData);
```

Comment:
@JGiter For `publishPublicClaim()` you instantiated `DIDDocumentFull` with the `ProxyOperator` to make the update. https://github.com/energywebfoundation/iam-client-lib/blob/b0d9446caf581500d084531a4917a2ea05c6d0dc/src/iam.ts#L369 Why is that?